nsot/models.py

```diff
@@ -1470,6 +1470,105 @@ def to_dict(self):
         }
 
 
+class Circuit(Resource):
+    """Represents two network Interfaces that are connected"""
+
+    # a_endpoint interface
+    a_endpoint = models.ForeignKey(
+        Interface, db_index=True, related_name='circuits_a',
+        null=False, verbose_name='A-side Interface',
+        help_text='Unique ID of interface at the A-side'
+    )
+
+    # z_endpoint interface
+    z_endpoint = models.ForeignKey(
+        Interface, db_index=True, related_name='circuits_z',
+        null=True, verbose_name='Z-side Interface',
+        help_text='Unique ID of interface at the Z-side'
+    )
+
+    # We are currently inferring the site_id from the parent (A-side) Interface
+    # in the .save() method
+    site = models.ForeignKey(
+        Site, db_index=True, related_name='circuits',
+        on_delete=models.PROTECT,
+        help_text='Unique ID of the Site this Circuit is under'
+    )
+
+    name = models.CharField(
+        max_length=255, unique=True,
+        help_text='Unique display name of circuit',
+        default='',
+    )
+
+    def __unicode__(self):
+        return u'%s' % self.name
+
+    class Meta:
+        # An interface shouldn't be able to be part of multiple circuits. This
+        # fixes the problem... mostly
+        #
+        # This doesn't solve an interface from being in one a_endpoint and
+        # another z_endpoint, but solves the others
+        unique_together = (
+            ('a_endpoint', 'z_endpoint'),
+            ('a_endpoint',),
```

Comment:
What's your motivation for this constraint? It would be better represented as a `OneToOneField` relationship for each endpoint which is effectively a `ForeignKey` with a unique constraint but which only returns a single object on either side vs. one-to-many.